transcom/mymove
# INT B-22057

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-22057)

## Summary

With the work done in [B-21986](https://github.com/transcom/mymove/pull/14317) we changed how the estimated & (new) max incentive calculations are done for PPMs. Problem is, we had a handful of PPMs in prod that need to be retroactively addressed.

This PR added a prod migration that will change move, shipment, and PPM statuses when:
- `max_incentive` in `ppm_shipments` is `NULL`

This PR also adds a way for the SC to see the `Max Incentive` value stored in the `max_incentive` column in the `ppm_shipments` table

Migration:
```
-- if the PPM has a null max incentive, we need to let the SC go through the workflow to recalculate the max/estimated incentives
-- change move status so SCs can know to take these actions
UPDATE moves
SET status = 'NEEDS SERVICE COUNSELING'
FROM mto_shipments
JOIN ppm_shipments ON mto_shipments.id = ppm_shipments.shipment_id
WHERE ppm_shipments.max_incentive IS NULL
  AND mto_shipments.move_id = moves.id;

UPDATE mto_shipments
SET status = 'SUBMITTED'
WHERE id IN (
    SELECT mto_shipments.id
    FROM mto_shipments
    JOIN ppm_shipments ON mto_shipments.id = ppm_shipments.shipment_id
    WHERE ppm_shipments.max_incentive IS NULL
);

UPDATE ppm_shipments
SET status = 'SUBMITTED'
WHERE max_incentive IS NULL;

UPDATE ppm_shipments
SET advance_status = NULL
WHERE max_incentive IS NULL
    AND has_requested_advance = 'true';
```
This will land the PPM/shipments back into the SC's queue, where they can "Submit move details" again after reviewing and the max and estimated incentives will recalculate based on the new code in `pkg/services/move_task_order/move_task_order_updater.go`

### How to test

1. Access MM as a customer
2. Complete PPMs with & without an advance and submit the move
3. Complete the approval process as the SC
4. Find the PPMs in DBeaver and turn the `max_incentive` value to `NULL`
5. Then run the above migration snippet in DBeaver
6. Navigate back to the moves (which should be in `NEEDS SERVICE COUNSELING` status)
7. If you are approving an advance, you will see the `max_incentive` populate after saving & continuing through that
8. If you did NOT request an advance, that column will populate when you click `Submit move details` as the SC

<img width="855" alt="Screenshot 2024-12-04 at 3 21 48 PM" src="https://github.com/user-attachments/assets/dccc70a5-2138-451f-8fd4-5db0c56cd852">
